test/functional/api/buffer_updates_spec.lua

```diff
@@ -741,3 +742,90 @@ describe('API: buffer events:', function()
   end)
 
 end)
+
+describe('API: buffer events:', function()
+  before_each(function()
+    helpers.clear()
+  end)
+
+  local function lines_subset(first, second)
+    for i = 1,#first do
+      if first[i] ~= second[i] then
+        return false
+      end
+    end
+    return true
+  end
+
+  local function lines_equal(f, s)
+    return lines_subset(f, s) and lines_subset(s, f)
+  end
+
+  local function did_match_somewhere(buffer_lines, expected_lines)
```

Comment:
I renamed this to `assert_match_somewhere` and swapped the parameter order (typically the "expected" value is the first parameter).